test/e2e/e2e_test.go

```diff
@@ -292,6 +299,134 @@ func pollForConfigMapChange(client *kubernetes.Clientset, compareConfigMap *v1.C
 	})
 }
 
+// createRouteForService creates a OpenShift routes
+func createRouteForService(routeClient *routeclient.RouteV1Client, s *v1.Service, namespace string) (*routev1.Route, error) {
```

Comment:
Exposing the metrics endpoint for a route should not be necessary here - way too complicated. The endpoint should be checked from inside the cluster

Reply:
I think that sounds reasonable to me. I have designed this logic after having some discussion with @stlaz 
@stlaz wdyt?

Reply:
either is fine I think, I noticed people from console doing this and it looked like a good idea to me, for testing

Reply:
A common pattern in upstream kube when needing to validate a cluster-internal endpoint is to launch a busybox pod that connects to the endpoint. This pattern limits the scope of test dependency. Why is it desirable for validation of internal connectivity to depend on external connectivity?